zebrad/tests/acceptance.rs

```diff
@@ -3127,3 +3128,30 @@ async fn validate_regtest_genesis_block() {
         "validated block hash should match network genesis hash"
     )
 }
+
+/// Test that Version messages are sent with the external address when configured to do so.
+#[test]
+fn external_address() -> Result<()> {
+    let _init_guard = zebra_test::init();
+    let testdir = testdir()?.with_config(&mut external_address_test_config(&Mainnet)?)?;
+    let mut child = testdir.spawn_child(args!["start"])?;
+
+    // Give enough time to start connecting to some peers.
+    std::thread::sleep(Duration::from_secs(10));
+
+    child.kill(false)?;
+
+    let output = child.wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    // Zebra started
+    output.stdout_line_contains("Starting zebrad")?;
+
+    // Make sure we are using external address for Version messages.
+    output.stdout_line_contains("Using external address for Version messages")?;
+
+    // Make sure the command was killed.
+    output.assert_was_killed()?;
+
+    Ok(())
+}
```

Comment:
Could this be replaced with a test that checks the version message, similar to the `connect_isolated_sends_anonymised_version_message_tcp()` test?

Reply:
It's a good idea but i think this one is enough for now.